go/vt/vtgateproxy/mysql_server.go

```diff
@@ -377,6 +360,23 @@ func (ph *proxyHandler) getSession(ctx context.Context, c *mysql.Conn) (*vtgatec
 	return session, nil
 }
 
+func (ph *proxyHandler) closeSession(ctx context.Context, c *mysql.Conn) {
+	session, _ := c.ClientData.(*vtgateconn.VTGateSession)
+	if session == nil {
+		return // no active session
```

Comment:
Previously when this was in ::getSession, it would create the session, which would trigger the spurious error if the Conn had no attributes because it wasn't properly established in the first place.